### PR TITLE
Duplicates

### LINKED ISSUE #209: Unimported operations
Here is my use case : 
- on the same day I did exactly the same operation (same amount, same type, same payee etc) 
- I think I merged the duplicates (this is my fault I know), but I am not 100% sure

The reimport is not possible because checks first if the operations are allLike before importing (processRetrievedOperation) 

Maybe using http://dev.weboob.org/api/capabilities/bank.html?highlight=unique_id#weboob.capabilities.bank.Transaction.unique_id from weboob API. (but to me this won't work)

### PR DESCRIPTION
**Pull request by ZeHiro**
Fixes #209.
And improves fake operation generation : 
- Generation of duplicates with +/- 1 day
- Generation of twice the same operation on the same import

### CODE REVIEW

File: server/lib/accounts-manager.js [hunk 1]
```diff
@@ -199,14 +199,20 @@ export default class AccountManager {
...
         for (let operation of operations) {
             let similarOperations = await Operation.allLike(operation);
             if (similarOperations && similarOperations.length)
                 continue;
+            operationsToCreate.push(operation);
+        }
 
+        // Create the new operations
+        for (let operationToCreate of operationsToCreate) {
             log.info('New operation found!');
```
Comment (bnjbvr) [inline server/lib/accounts-manager.js]: As we now have this information, can you:

`log.info(${operationToCreate.length} new operations found!`

if there are any, please? It's just much better for the logging.

File: server/lib/sources/mock.js [hunk 2]
```diff
@@ -184,6 +184,34 @@ let generate = uuid => {
         count++;
     }
 
+    // Generate exact same operations imported at the same time
+    // These operations shall not be considered as duplicates.
+    if (rand(0, 100) > 85) {
```
Comment (bnjbvr) [inline server/lib/sources/mock.js]: nit: add to the condition `&& operations.length`

File: server/lib/sources/mock.js [hunk 3]
```diff
@@ -184,6 +184,34 @@ let generate = uuid => {
...
+    if (rand(0, 100) > 70) {
+        let duplicateOperation = {
+            label: 'This is a duplicate operation',
+            amount: '13.37',
+            raw: 'This is a duplicate operation',
+            account: hashAccount(uuid).main
+        };
+        // The rdate is one day difference, so it is considered a duplicate by
+        // the client
+        let date = moment(new Date('05/04/2020'));
```
Comment (bnjbvr) [inline server/lib/sources/mock.js]: Can you put it in the past rather than in the future, please?

**Comment by bnjbvr** [comment]
Very nice idea to make the fake account generator create duplicate operations! Thank you for this PR. A few things to fix before it can land.

File: server/lib/sources/mock.js (hunk 3 above)
Comment (ZeHiro) [inline server/lib/sources/mock.js]: The idea is to have it "on top" in the report view, and test more easily the duplicates.

Comment (bnjbvr) [inline server/lib/sources/mock.js]: OK.

File: server/lib/accounts-manager.js (hunk 1 above)
Comment (ZeHiro) [inline server/lib/accounts-manager.js]: OK

File: server/lib/sources/mock.js (hunk 2 above)
Comment (ZeHiro) [inline server/lib/sources/mock.js]: OK

**Comment by bnjbvr** [comment]
Looks perfect, cheers!

**Pull request merged**